### PR TITLE
CI: Switch SIMD tests to meson

### PR DESCRIPTION
**Pull request by seiko2plus**
All SIMD tests have been transitioned to utilize the Meson build system.
  The Intel SDE tests are now consolidated into a single action.
  Additionally, the Arm7 test has been removed as it is already
  covered by Linux QEMU tests for armhf.

### CODE REVIEW

**Review by charris — APPROVED** [review]

**Comment by charris** [comment]
LGTM, but I'll wait for another review.

File: .github/workflows/linux_simd.yml
Comment (rgommers) [inline .github/workflows/linux_simd.yml:101]: This step does not seem to work, the output is only `^@` and afterwards the build log still shows:
```
C compiler for the host machine: cc (gcc 8.4.0 "cc (Ubuntu 8.4.0-3ubuntu2) 8.4.0")
```

File: .github/meson_actions/action.yml
Comment (rgommers) [inline .github/meson_actions/action.yml:24]: This step being included with all the others results in log browsing being a bit slow, since it's >10,000 lines as a result of `meson-log.txt` being so large. It's hidden by default though, and I think this is fine as a price to pay for such a concise `linux_simd.yml` file. So I wanted to note this, but no change needed.

**Review by rgommers — COMMENTED** [review]
This looks great to me, thanks @seiko2plus! I only have one comment about GCC 9 installation not being successful.

File: .github/workflows/linux_simd.yml
Comment (rgommers) [inline .github/workflows/linux_simd.yml:165]: For completeness: a superset of this test is now tested in `linux_qemu.yml`, so this removal is expected.

File: .github/workflows/linux_simd.yml
Comment (r-devulap) [inline .github/workflows/linux_simd.yml:197]: This was meant to show CPU info of the machine the CI job landed on (was useful to figure out the bug in SDE). 

```suggestion
        python -c "import numpy; numpy.show_config()" &&
```

**Review by rgommers — APPROVED** [review]
LGTM now, thanks @seiko2plus!

File: .github/workflows/linux_simd.yml
Comment (rgommers) [reply .github/workflows/linux_simd.yml:197]: The `sde -spr` usage doesn't seem needed to me here indeed. But it'll work either way, the output in the CI log seems fine. Does that sound right? If so, I can take this along in a follow-up (need to make more CI tweaks soon).

**Pull request merged**
